homeassistant/components/samsungtv/media_player.py

```diff
@@ -52,11 +56,27 @@
 )
 
 
+@core.callback
+def async_get_on_script_from_mac(hass, mac):
+    """Create an on script from a mac address."""
+    script = cv.SCRIPT_SCHEMA(
+        {
+            CONF_SERVICE: "wake_on_lan.send_magic_packet",
+            CONF_SERVICE_DATA: {CONF_MAC: mac},
+        }
+    )
+    return Script(hass, script)
+
+
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the Samsung TV from a config entry."""
     ip_address = config_entry.data[CONF_IP_ADDRESS]
     on_script = None
-    if (
+
+    if CONF_MAC in config_entry.options:
```

Comment:
Not sure how to test this part. If I understand correctly, this function is currently not covered by tests.

Reply:
@MartinHjelmare this is where coverage is impacted. It's definitely logic worth testing, but I don't believe this function is tested at all, and not sure how to go about it. Any help would be appreciated.

Reply:
@MartinHjelmare any guidance? thanks